src/aspire/source/xform.py

```diff
@@ -150,6 +150,26 @@ def _adjoint(self, im, indices):
         return im.shift(-self.shifts[indices])
 
 
+class Reduce(LinearXform):
```

Comment:
`Reduce` is a bit ambiguous here (has connotations related to map–reduce and friends). Why not `Add`? It's also not a linear transform (so it doesn't have an adjoint).